### ISSUE TITLE
JS file with default config gets convert-to-es6-module suggestion

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#28362]
**TypeScript Version:**  3..2.0-dev.20181103

**Search Terms:**

**Code**

```ts
exports.x = 0;
```

**Expected behavior:**

No suggestion, because there is no tsconfig.json or jsconfig.json with compiler options expressing an intent to use ES6 modules.

**Actual behavior:**

There is a suggestion diagnostic to convert to an ES6 module.
I added a log in `suggestionDiagnostics.ts` to print the config it had:
`{"module":1,"target":3,"jsx":1,"allowJs":true,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true,"noEmitForJsFiles":true,"maxNodeModuleJsDepth":2}`

Since the compilerOptions are sufficient to cause the suggestion, I'm pretty sure this has nothing to do with ATA.
The default config apparently has the `module` option set, which makes us think the user must be transpiling modules and so ES6 modules are a good idea.

I can't seem to find where this default config is specified -- @sheetalkamat do you know?

I also wrote this test to try and reproduce it, but it doesn't happen in the test.
```ts
describe("tsserverProjectSystem convert to es6 module", () => {
    it("does not trigger without a tsconfig", () => {
        const aJs: File = { path: "/a.js", content: "exports.x = 0;" };
        const session = createSession(createServerHost([aJs]));
        openFilesForSession([aJs], session);
        const response = executeSessionRequest<protocol.SuggestionDiagnosticsSyncRequest, protocol.SuggestionDiagnosticsSyncResponse>(
            session, protocol.CommandTypes.SuggestionDiagnosticsSync, { file: aJs.path });
        assert.deepEqual<ReadonlyArray<protocol.DiagnosticWithLinePosition | protocol.Diagnostic> | undefined>(response, []);
    });
});
```

**Comment by sheetalkamat** [comment]
It might be because the inferred project gets options set by `setCompilerOptionsForInferredProject` command.

**Comment by andy-ms** [comment]
The editor doesn't seem to be sending that command.

**Comment by andy-ms** [comment]
Hm, `function getDefaultCompilerOptions` sets `target: ScriptTarget.ES5` which is `1`, while I saw `3` which corresponds to ES2016.
By the way, all you need to test this is to add `if ((true)) throw new Error(JSON.stringify({ options: program.getCompilerOptions() }));` to the top of `function computeSuggestionDiagnostics`.

**Comment by sheetalkamat** [comment]
In vscode I always see compiler options for inferred project so that would explain why you see it?
```
Info 11   [8:50:15.496] request:
    {"seq":1,"type":"request","command":"compilerOptionsForInferredProjects","arguments":{"options":{"module":"commonjs","target":"es2016","jsx":"preserve","allowJs":true,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true}}}
```

**Comment by andy-ms** [comment]
That would explain it. I was searching for `setCompiler...` before and didn't see that.
I'm thinking we should ask vscode to use `moduleResolution` and `lib` options instead of `module` and `target`? `module` and `target` indicate how the code should be *emitted*, so we use them as an indication that code is being transpiled and it's OK to use ES6 module systax. `moduleResolution` and `lib` only affect how it is checked.

**Comment by sheetalkamat** [comment]
I think target is used to stop unnecessary getter cannot be used kind of errors?

**Comment by andy-ms** [comment]
That could work with `--target es5` at least -- async functions don't show an error for that target either.

**Comment by sheetalkamat** [comment]
@DanielRosenwasser Is this important enough scenario and do we want default target to be es5 on compiler options or --es6. Vscode sets it through `compilerOptionsForInferredProjects` command. Adding @mjbvz as well.
